docs/dev/rfc/odfe-cli-rest-api-as-commands.md

```diff
@@ -0,0 +1,298 @@
+|--human	| N/A	| false	|the output statistics are returned in a human friendly format. (e.g. `"exists_time": "1h"` or `"size": "1kb"`) instead of (e.g. `"exists_time_in_millis": 3600000` or `"size_in_bytes": 1024`). 	|
+|--filter-path	|-f	| N/A	|filter the response fields returned by cluster. It accepts a comma separated list of filters expressed with the dot notation.  eg: `--filter-path=“took,hits.hits._id,hits.hits._score”`	|
+
+
+
+## 1. get command
+
+### Description
+
+Use GET API to execute requests against elasticsearch cluster. This command enables you to run any GET based REST API commands across all 
+categories.
+
+### Synopsis
+
+```
+odfe-cli cmd get options [[common parameters]](https://quip-amazon.com/XKAiAJe5eP1U/RFC-Support-REST-api-in-odfe-cli#eBP9CAENWYx)
+```
+
+### Options
+
+|Name|Required|Short hand|Usage|Separator for multiple values|
+|---| :---: |:---:|---	|:---:|
+| --path	|Y	|-P	|url-path	|N/A	|
+| --query-params|N	|-q	|url query parameters	|&	|
+| --headers	 |N	|-H	|pass additional information with request. It consists of case-insensitive name followed by a colon (`:`), then by its value.	|semi colon ( ; )	|
+|--data	|N	|-d	|Sends the specified data in the command to server". If value starts with the letter @, the rest should be a file name to read the data from.	|N/A	|
+|--data-binary	|N	|-b	|This is similar to "--data", except that newlines and carriage returns are preserved and conversions are never done. This is required if input file is compressed.	|N/A	|
+|--help	|N	|-h	|Help for get command	|N/A	|
+
+
+**Note: users can escape separator like ‘\\;’ to use as part of value.**
+
+### Example:
+
+1. To get document count for an index
+
+```
+odfe-cli cmd get --path "_cat/count/my-index-01" \
+                 --query-params "v=true" \
+                 --pretty
+```
+
+Response
+
+```
+epoch         timestamp        count
+1612311592    12:24:24          100
+```
+2.  To  returns the health status of a cluster.
+
+```
+odfe-cli cmd get --path "_cluster/health" --pretty
+```
+
+Response
+
+```
+{
+    "cluster_name" : "odfe-cli",
+    "status" : "green",
+    "timed_out" : false,
+    "number_of_nodes" : 2,
+    "number_of_data_nodes" : 2,
+    "active_primary_shards" : 1,
+    "active_shards" : 1,
+    "relocating_shards" : 0,
+    "initializing_shards" : 0,
+    "unassigned_shards" : 0,
+    "delayed_unassigned_shards": 0,
+    "number_of_pending_tasks" : 0,
+    "number_of_in_flight_fetch": 0,
+    "task_max_waiting_in_queue_millis": 0,
+    "active_shards_percent_as_number": 100.0
+}
+```
+
+ 3.  To explain cluster allocation for index “my-index-01”, shard ‘0’
+
+
+```
+odfe-cli cmd get --path "_cluster/allocation/explain" \
+                 --data  '{
+                    "index": "my-index-01",
+                    "shard": 0,
+                    "primary": false,
+                    "current_node": "nodeA"                         
+                  }'
+```
+
+## 2. post command
```

Comment:
Should post be capitalized?

Reply:
Here it is not a api, it is a command. Since it will not be capitalized in ur tool, i kept it as it is for consistency. What do you think?

Reply:
Makes sense